src/IO/H5/Python/File.cpp

```diff
@@ -17,32 +17,73 @@ void bind_h5file(py::module& m) {  // NOLINT
   // Wrapper for basic H5File operations
   using h5_file_rw = h5::H5File<h5::AccessType::ReadWrite>;
   py::class_<h5_file_rw>(m, "H5File")
-      .def(py::init<std::string, bool>(), py::arg("file_name"),
-           py::arg("append_to_file") = false)
+      .def(py::init([](const std::string& file_name, bool append_to_file) {
+             const bool file_exists =
+                 file_system::check_if_file_exists(file_name);
+             if (file_exists && !append_to_file) {
+               throw std::runtime_error("File " + file_name +
+                                        " already exists. To append to it, "
+                                        "open with `append_to_file` = True.");
+             }
+             return h5_file_rw{file_name, append_to_file};
+           }),
+           py::arg("file_name"), py::arg("append_to_file") = false)
       .def("name", &h5_file_rw::name)
       .def(
           "get_dat",
           +[](const h5_file_rw& f, const std::string& path) -> const h5::Dat& {
+            if (not f.exists<h5::Dat>(path)) {
+              auto groups = f.groups();
+              std::string subfiles{};
+              for (auto&& group : groups) {
+                subfiles += group + " ";
+              }
+              throw std::runtime_error(
+                  "Subfile `" + path + "` was not found in file `" + f.name() +
+                  "`. Available subfiles are:\n" + subfiles);
```

Comment:
I suggest to either use `boost::algorithm::join` to create the comma-delimited string, or just use `get_output(groups)` (from `Utilities/GetOutput.hpp`) to get some representation of the vector that's probably good enough